waku/v2/peermanager/peer_manager.go

```diff
@@ -87,6 +102,55 @@ func inAndOutRelayPeers(relayPeers int) (int, int) {
 	return relayPeers - outRelayPeers, outRelayPeers
 }
 
+// checkAndUpdateTopicHealth finds health of specified topic and updates and notifies of the same.
+// Also returns the healthyPeerCount
+func (pm *PeerManager) checkAndUpdateTopicHealth(topic *NodeTopicDetails) int {
+	healthyPeerCount := 0
+	for _, p := range topic.topic.ListPeers() {
+		if pm.host.Network().Connectedness(p) == network.Connected {
+			pThreshold, err := pm.host.Peerstore().(wps.WakuPeerstore).Score(p)
+			if err == nil && pThreshold < relay.PeerPublishThreshold {
+				pm.logger.Debug("peer score below publish threshold", logging.HostID("peer", p), zap.Float64("score", pThreshold))
+			} else {
+				//For now considering peer as healthy if we can't fetch score.
+				healthyPeerCount++
+			}
+			if err != nil {
+				pm.logger.Warn("failed to fetch peer score ", zap.Error(err), logging.HostID("peer", p))
+			}
```

Comment:
I find this somewhat confusing, what do you think if we change it like this
```suggestion
			if err == nil {
			    if pThreshold < relay.PeerPublishThreshold {
				pm.logger.Debug("peer score below publish threshold", logging.HostID("peer", p), zap.Float64("score", pThreshold))
			    } else {
				    //For now considering peer as healthy if we can't fetch score.
				    healthyPeerCount++
			    }
		       } else {
				pm.logger.Warn("failed to fetch peer score ", zap.Error(err), logging.HostID("peer", p))
			}
```

Reply:
Looks like i missed handling a case while merging conditions in order to optimize. 
Let me clean it up. 

My idea was to consider a peer as healthy even if we can't fetch its score and increase healthyPeerCount.